operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,916 @@
+        //   0,   0, 255,   0,   0  ||   100,   0,   0, 255, 100  ||  0, 60, 15,  5,  0
+        //   0, 255,   0, 255,   0  ||   100,   0,   0,   0, 100  ||  0, 15, 60, 15,  0
+        //   0,   0, 255,   0,   0  ||   100, 255,   0,   0, 100  ||  0,  5, 15, 60,  0
+        //   0,   0,   0,   0,   0  ||   100, 100, 100, 100, 100  ||  0,  0,  0,  0,  0
+
+        // with class layout like this:
+        // 3, 3, 3, 3, 3
+        // 3, 2, 0, 1, 3
+        // 3, 0, 2, 0, 3
+        // 3, 1, 0, 2, 3
+        // 3, 3, 3, 3, 3
+
+        let b1 = &[
+            0., 0., 0., 0., 0., 0., 0., 255., 0., 0., 0., 255., 0., 255., 0., 0., 0., 255., 0., 0.,
+            0., 0., 0., 0., 0.,
+        ];
+
+        let b2 = &[
+            100., 100., 100., 100., 100., 100., 0., 0., 255., 100., 100., 0., 0., 0., 100., 100.,
+            255., 0., 0., 100., 100., 100., 100., 100., 100.,
+        ];
+
+        let b3 = &[
+            0., 0., 0., 0., 0., 0., 60., 15., 5., 0., 0., 15., 60., 15., 0., 0., 5., 15., 60., 0.,
+            0., 0., 0., 0., 0.,
+        ];
+
+        let test_data_arr_2d = arr2(zip_bands(b1, b2, b3).as_slice());
+        let test_data = make_xg_matrix(&test_data_arr_2d);
+        let result = bst.predict(&test_data).unwrap();
+
+        // this result is not 100% desired, but most likely just a matter of a better or bigger training set
+        let expected = vec![
+            3.0, 3.0, 3.0, 3.0, 3.0, 3.0, 0.0, 0.0, 3.0, 3.0, 3.0, 0.0, 0.0, 0.0, 3.0, 3.0, 3.0,
+            0.0, 0.0, 3.0, 3.0, 3.0, 3.0, 3.0, 3.0,
+        ];
+
+        assert_eq!(expected, result);
+    }
+
+    /// This test verifies, that xgboost is creating meaningful result tiles from a learned model.
+    #[tokio::test]
+    async fn xg_single_tile_test() {
+        // 255, 255,   0,   0,   0
+        // 255, 255,   0,   0,   0
+        //   0,   0,   0,   0,   0
+        //   0,   0,   0, 255, 255
+        //   0,   0,   0, 255, 255
+        let green = make_raster(vec![
+            255, 255, 0, 0, 0, 255, 255, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 0, 0, 0, 255,
+            255,
+        ]);
+
+        //   0,   0,   0, 255, 255
+        //   0,   0,   0, 255, 255
+        //   0,   0,   0,   0,   0
+        // 255, 255,   0,   0,   0
+        // 255, 255,   0,   0,   0
+        let blue = make_raster(vec![
+            0, 0, 0, 255, 255, 0, 0, 0, 255, 255, 0, 0, 0, 0, 0, 255, 255, 0, 0, 0, 255, 255, 0, 0,
+            0,
+        ]);
+
+        //   15,  15,  60,   5,   5
+        //   15,  15,  60,   5,   5
+        //   60,  60,  60,  60,  60
+        //    5,   5,  60,  15,  15
+        //    5,   5,  60,  15,  15
+        let temp = make_raster(vec![
+            15, 15, 60, 5, 5, 15, 15, 60, 5, 5, 60, 60, 60, 60, 60, 5, 5, 60, 15, 15, 5, 5, 60, 15,
+            15,
+        ]);
+
+        let srcs = vec![green, blue, temp];
+
+        let model_path = PathBuf::from("pro/ml/xgboost/s2_10m_de_marburg/model.json");
```

Comment:
please use the test_data! macro here (and in the other tests). otherwise the data may not be found, as the working dir is sometimes different, depending on how the tests are run.

Reply:
fd940cc820ae7165086bbe9e41f6af5fedf83830